src/pds/provenance/provenance.py

```diff
@@ -0,0 +1,269 @@
+#   Laboratory, nor the names of its contributors may be used to endorse or
+#   promote products derived from this software without specific prior written
+#   permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# provenance
+# ==========
+#
+# Determines if a particular document has been superseded by a more
+# recent version, if upon which it has, sets the field 
+# ops:Provenance/ops:superseded_by to the id of the superseding document.
+#
+# It is important to note that the document is updated, not any dependent
+# index.
+#
+
+import argparse
+import collections
+import json
+import logging
+from typing import Union, List, Mapping, Iterable
+
+log = logging.getLogger('provenance')
+import requests
+import urllib.parse
+
+requests.packages.urllib3.disable_warnings()
+
+HOST = collections.namedtuple('HOST', ['nodes', 'password', 'url', 'username',
+                                       'verify'])
+
+METADATA_SUCCESSOR_KEY = 'ops:Provenance/ops:superseded_by'
+
+
+def parse_log_level(input: str) -> int:
+    """Given a numeric or uppercase descriptive log level, return the associated int"""
+    try:
+        result = int(input)
+    except ValueError:
+        result = getattr(logging, input)
+    return result
+
+
+def _vid_as_tuple_of_int(lidvid: str):
+    M, m = lidvid.split('::')[1].split('.')
+    return (int(M), int(m))
+
+
+def configure_logging(filepath: Union[str, None], log_level: int):
+    logging.root.handlers = []
+    handlers = [
+        logging.StreamHandler()
+    ]
+
+    if filepath:
+        handlers.append(logging.FileHandler(filepath))
+
+    logging.basicConfig(
+        level=log_level,
+        format='%(asctime)s::%(levelname)s::%(message)s',
+        handlers=handlers
+    )
+
+
+def run(
+        cluster_nodes: List[str],  # TODO: confirm type
+        base_url: str,
+        username: str,
+        password: str,
+        verify_host_certs: bool = False,
+        log_filepath: Union[str, None] = None,
+        log_level: int = logging.INFO):
+    configure_logging(filepath=log_filepath, log_level=log_level)
+
+    log.info('starting CLI processing')
+
+    host = HOST(cluster_nodes, password, base_url, username, verify_host_certs)
```

Comment:
Oh so that's how you use named tuple 😄

Reply:
This line was my first time too - thanks Al!